### PR TITLE
feat: [CDS-74693]: Add status icon in the execution id dropdown

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Add status icon in the execution id dropdown 
- Add clear btn to the execution ID field
- Prefixed `Execution ID :` in the label of the field

#### Screenshots

Screenshot 2023-07-20 at 4 47 23 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 1]
```diff
@@ -44,32 +45,33 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
       repoIdentifier,
       branch,
       parentEntityConnectorRef: storeMetadata?.connectorRef,
-      parentEntityRepoName: storeMetadata?.repoName
+      parentEntityRepoName: storeMetadata?.repoName,
+      page: 0,
+      size: 20
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:50]:
```suggestion
      size: DEFAULT_PAGE_SIZE
```

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 2]
```diff
@@ -44,32 +45,33 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
...
+          ...iconMap[execId.status as ExecutionStatus],
+          className: cx(css.status, css[(execId.status as ExecutionStatus).toLowerCase() as keyof typeof css], css.icon)
+        }
       }
     }),
     []
   )
 
-  executionIdSelectOptions.push(noExecution)
+  // executionIdSelectOptions.push(noExecution)
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:69]: Do we not thus empty state anymore?

Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:69]: If that is not needed we can remove this commented code

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 3]
```diff
@@ -108,9 +110,13 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
           items={executionIdSelectOptions}
           onChange={handleExecutionIdChange}
           name="executionId"
-          placeholder="ExecutionID"
-          value={executionIdOption}
+          placeholder="Select Execution Id"
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:113]: This should comes from yaml

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 4]
```diff
@@ -108,9 +110,13 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
...
+          value={{
+            label: executionIdOption?.label ? `Execution ID: ${executionIdOption.label}` : '',
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx:115]: Same as above

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx [hunk 5]
```diff
@@ -44,32 +45,33 @@ export function VariablesCompiledModeHeader(props: VariablesCompiledModeHeaderPr
       repoIdentifier,
       branch,
       parentEntityConnectorRef: storeMetadata?.connectorRef,
-      parentEntityRepoName: storeMetadata?.repoName
+      parentEntityRepoName: storeMetadata?.repoName,
+      page: 0,
```
[the remark this answers is not in the document]
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: Removed

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: done

File: src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx (hunk 4 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineVariables/VariablesHeader/VariablesCompiledModeHeader.tsx]: done

**Review by pankajsingh-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

**Pull request merged**